Lombiq.Privacy/Startup.cs

```diff
@@ -43,6 +43,20 @@ public override void ConfigureServices(IServiceCollection services)
             options.MinimumSameSitePolicy = SameSiteMode.Strict;
             options.Secure = CookieSecurePolicy.Always;
             options.ConsentCookie.Expiration = new TimeSpan(365, 0, 0, 0);
+
+            options.OnAppendCookie = cookieContext =>
+            {
+                // Check if the cookie name matches the ones used for Azure AD authentication.
+                if (cookieContext.CookieName.Contains("AspNetCore.OpenIdConnect.Nonce")
+                    || cookieContext.CookieName.Contains("AspNetCore.Correlation")
+                    || cookieContext.CookieName.Contains("Identity.External"))
+                {
+                    // For Azure AD to work with Lombiq.PrivacyPolicy, we need to set the CookieOptions.SameSite value
+                    // to SameSiteMode.None. Without this the Azure AD authentication will fail with the
+                    // "System.Exception: Correlation failed" error.
+                    cookieContext.CookieOptions.SameSite = SameSiteMode.None;
```

Comment:
Isn't it possible to somehow tie the cookie to only certain domains? Because this way, it's shared around, but can't it be just shared with the URLs necessary for Azure AD login?

Reply:
We can check the path but I think that one can change with the callback URL (it's `~/signin-oidc` by default) `cookieContext.CookieOptions.Path.Contains("signin-oidc") `. However this way we also need to check this cookie `cookieContext.CookieName.Contains("Identity.External")` since it's needed and it's not on that URL.

Reply:
I'm not talking about paths under the current domain, but restricting other domains. Because with `SameSiteMode.None`, the cookie can be accessed on any site that we don't control.

Reply:
I see. I think it's possible with `cookieContext.CookieOptions.Domain`.

Reply:
Please check then if we can use that to allow only the current request's domain and those from Microsoft that are necessary for this to work.

Reply:
> and those from Microsoft that are necessary for this to work

I checked those for these cookies, but I haven't found any Microsoft ones that were needed. It was working with the current request's domain (so localhost or in the code `cookieContext.CookieOptions.Domain = cookieContext.Context.Request.Host.Host;`)